src/shared.rs

```diff
@@ -141,9 +143,9 @@ pub struct BackendRef {
     /// Support for this field varies based on the context where used.
     pub weight: Option<u16>,
 
-    pub name: String,
-
-    pub port: PortNumber,
+    /// BackendObjectReference references a Kubernetes object.
+    #[serde(flatten)]
+    pub inner: BackendObjectReference,
```

Comment:
maybe we should give this a more descriptive name, like `backend`?
```suggestion
    pub backend: BackendObjectReference,
```

Reply:
not saying this is good or bad, but there's precedent for using the name `inner` for this type of thing:

https://github.com/linkerd/k8s-gateway-api-rs/blob/main/src/httproute.rs#L27

Reply:
Yeah, I'm also easy. I wanted to use `backend_ref` at first. I'm okay with `backend` too if we don't care that much about the precedent.

Reply:
🤷‍♀️ not a strongly held opinion. wouldn't be the first `inner`!

Reply:
Hmm will keep it the same, think we have some more `inner` values in there :( just for consistency, but wouldn't be opposed to changing this everywhere later as a cosmetic thing.